### ISSUE TITLE
When enlisted with distributor `SendLocal` sends a message back to the worker's queue

### ISSUE THREAD

**Issue by SzymonPobiega** [Particular/NServiceBus#4338]
### Who's affected

 * Users who upgraded their distributor workers to V6

### Symptoms

Several API calls that should result in routing a message back to the distributor do not work correctly (and route the message to the local worker queue):
 * `IMessageSession.SendLocal(new MyMessage())`
 * Sending a delayed message to the local endpoint:

```
var options= new SendOptions();
options.DelayDeliveryWith(delay);
options.RouteToThisEndpoint();
session.Send(new MyMessage(), options);
```
 * Delaying a message that failed all FLR attempts in SLR

**Comment by timbussmann** [comment]
is there any potential damage in handling the message on the current instance? I don't think there is a real need to address this so far.

**Comment by SzymonPobiega** [comment]
@timbussmann there is no damage but this is a different behavior than in V5 and different that the intention was when building the legacy distributor support for V6.

**Comment by SzymonPobiega** [comment]
It also does not play that well with the load balancing aspect of the distributor.

**Comment by SzymonPobiega** [comment]
This has been fixed in #4329

**Comment by timbussmann** [comment]
sidenote: this behavior only sends back the message to the distributor if `SendLocal` is called from a message handler handling a message dispatched from a distributor.
